data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -141,7 +140,8 @@ static InputStream postToStreamingOutput(String url, String payload, long timeou
 
     private static DataServiceException createExceptionFromResponse(String url, HttpResponse httpResponse, Exception thrownException, String errorFromResponse) {
         if (httpResponse == null) {
-            return new DataServiceException(url, "POST failed to send request", thrownException);
+            return new DataServiceException(url, "POST failed to send request", thrownException,
+                    TriState.DONT_KNOW);
```

Comment:
I'm not sure about this DONT_KNOW option. The way I see it, if we don't know that it's permanent, we generally assume transient. And over time we refine which errors are considered permanent/transient.

In this case, if `httpClient.execute(request)` returns null, I'd consider that to be transient. But more generally, I can't think of many times when DONT_KNOW should be used instead of FALSE.

In the .NET code, I see:
`
public static bool IsPermanent(this IngestionErrorCode ingestionErrorCode)
`
The property Kusto.Data.Net.Http.OneApiError.BaseError.permanent is a simple bool.

In Kusto.Cloud.Platform.Utils.ExtendedException.IsTransient, DONTKNOW is an option, but it's only returned if the exception doesn't match any of the if/elses.